src/license_expression/__init__.py

```diff
@@ -599,40 +599,51 @@ def simple_tokenizer(self, expression):
                         sym = LicenseSymbol(key=sym_or_op)
                     yield Token(start, end, sym_or_op, sym)
 
-    def expressions(self, expression):
+    def dedup(self, expression):
         """
-        Return a de-dupulicated expression
+        Return a de-duplicated expression
         """
         exp = self.parse(expression)
         dedup_expression = ''
         expression_list = []
         for arg in exp.args:
-            print(arg)
             if isinstance(arg, (self.AND, self.OR)):
                 # Run this recursive function if there is another AND/OR expression
                 # and add the expression to the expression_list. 
-                expression_list.append(self.expressions(arg))
+                expression_list.append(self.dedup(arg))
             else:
-                # Get the license key from the expression.
-                exp_key = self.license_keys(arg)[0]
+                # Get the license key from the expression as a list.
+                exp_key = self.license_keys(arg)
+                # We treat the license with exception as a single license key so
+                # that it won't over de-dupped for case such as 
+                # gpl-2.0 WITH classpath exception AND gpl-2.0
+                if type(arg).__name__ == 'LicenseWithExceptionSymbol':
```

Comment:
@pombredanne I think you won't be happy with this line of code, but I am not sure how to check the class type here other than using the .__name__ 
I need direction :)

Reply:
You could have used isinstance() instead... That said, I have pushed a few refinements for your review.